### PR TITLE
2697: Remake ding lists

### PR DESCRIPTION
**Pull request by b14sha**
https://platform.dandigbib.org/issues/2697

### CODE REVIEW

File: modules/p2/ding_list/theme/theme.inc [hunk 1]
```diff
@@ -166,52 +160,92 @@ function template_preprocess_ding_list_list(&$variables) {
...
-          '#markup' => $list->getNote(),
+    default:
+      if ($list->isOwner()) {
+        $variables['content']['elements'] = array(
+          '#theme' => 'table',
+          '#attributes' => array(
+            'id' => drupal_html_id('ding-list-list-elements'),
+            'class' => array(
+              'ding-list-list__elements', 
+              'ding-list-list__owner',
```
Comment (kasperg) [inline modules/p2/ding_list/theme/theme.inc]: :+1: for the naming but I am confused. `ding-list-data--info` uses double `-`. This uses `_`. Why the change?

File: modules/p2/ding_list/theme/theme.inc [hunk 2]
```diff
@@ -166,52 +160,92 @@ function template_preprocess_ding_list_list(&$variables) {
...
-          'css' => array(
-            drupal_get_path('module', 'ding_list') . '/css/ding-list-tabledrag.css',
+      else {
+        $variables['content']['elements'] = array(
+          '#theme' => 'table',
+          '#attributes' => array(
+            'id' => drupal_html_id('ding-list-list-elements'),
+            'class' => array(
+              'ding-list-list__elements', 
+              'ding-list-list__guest',
```
Comment (kasperg) [inline modules/p2/ding_list/theme/theme.inc]: See comment regarding class naming above.

File: modules/p2/ding_list/theme/theme.inc [hunk 3]
```diff
@@ -308,5 +342,17 @@ function template_preprocess_ding_list_add_button(&$variables) {
...
+
+/**
+ * Ding list label/value theme.
+ */
+function theme_ding_list_label_value($variables) {
+  $label = '';
+  if ($variables['label']) {
+    $label = '<div class="label">' . $variables['label'] . '</div>';
+  }
+  $out = '<div class="field">' . $label . '<div class="text">' . $variables['value'] . '</div></div>';
```
Comment (kasperg) [inline modules/p2/ding_list/theme/theme.inc]: These class names are very generic. If this is to be considered a component (it has a separate theme function so I think that is a reasonable assumption) then I would expect it to be namespaced as such e.g `ding-list-field`, `ding-list-field-label` and `ding-list-field-text`.

File: modules/p2/ding_sharer/ding_sharer.module [hunk 4]
```diff
@@ -95,21 +96,28 @@ function template_preprocess_ding_sharer(&$variables) {
  */
 function theme_ding_sharer($variables) {
   $out = '';
+  $anchor_class = '';
 
-  if (!empty($variables['label'])) {
+  if (!empty($variables['label']) && $variables['type'] == 'normal') {
```
Comment (kasperg) [inline modules/p2/ding_sharer/ding_sharer.module]: This has normal (all lowercase). The default value is Normal (first letter uppercase). That does not seem right?

File: modules/p2/ding_sharer/ding_sharer.module [hunk 5]
```diff
@@ -95,21 +96,28 @@ function template_preprocess_ding_sharer(&$variables) {
...
   foreach ($variables['communities'] as $community => $text) {
-    $out .= l(
-      $text,
-      '#',
-      array(
-        'attributes' => array('class' => array($community . '-share')),
-        'external' => TRUE,
-      )
-    );
+    if ($text !== '0') {
```
Comment (kasperg) [inline modules/p2/ding_sharer/ding_sharer.module]: Again this comparison with the string `0`. Please avoid this by irrelevant communities beforehand.

File: themes/ddbasic/sass/components/ding-carousel.scss [hunk 6]
```diff
@@ -232,3 +232,17 @@
...
 }
+
+.ding-carousel-actions {
+  position: relative;
+  width: calc(100% - 166px);
+  bottom: 84px;
+  .more-link {
+    margin-right: 15px;
+  }
+  form {
```
Comment (kasperg) [inline themes/ddbasic/sass/components/ding-carousel.scss]: Why would there be a form inside `.ding-carousel-actions`? There does not seem to be any where it is defined in  `template_preprocess_ding_list_list()` as a part of this PR.

File: themes/ddbasic/sass/p2/p2-list.scss [hunk 7]
```diff
@@ -71,3 +72,9 @@
 .ding-list-list__title {
   @include font('display-large');
 }
+
+.ding-list-list__guest {
+  .ding-list-list__element-delta {
+    display: none;
+  }
+}
```
Comment (kasperg) [inline themes/ddbasic/sass/p2/p2-list.scss]: Nitpicking: Remember newlines.

File: themes/ddbasic/sass/p2/p2-view.scss [hunk 8]
```diff
@@ -42,3 +42,12 @@ tr.draggable {
     }
   }
 }
+
+// ==========================================================================
+// View ding public lists
+// ==========================================================================
+.pane-public-lists {
```
Comment (kasperg) [inline themes/ddbasic/sass/p2/p2-view.scss]: This seems to refer to a panel pane. Why is this not in `p2-class-panel.scss`?

File: modules/p2/ding_list/ding_list.info [hunk 9]
```diff
@@ -31,4 +32,4 @@ features[page_manager_pages][] = page_ding_list_create
 features[page_manager_pages][] = page_ding_list_edit
 features[page_manager_pages][] = page_ding_list_share
 features[page_manager_pages][] = page_ding_list_view
-# required = 1
+required = 1
```
[the remark this answers is not in the document]
Comment (b14sha) [reply modules/p2/ding_list/ding_list.info]: Has been removed

File: modules/p2/ding_list/ding_list.pages_default.inc [hunk 10]
```diff
@@ -563,7 +563,7 @@ function ding_list_default_page_manager_pages() {
     ),
   );
   $display->cache = array();
-  $display->title = '%ding_list_list:title';
+  $display->title = '';
```
[the remark this answers is not in the document]
Comment (b14sha) [reply modules/p2/ding_list/ding_list.pages_default.inc]: Has been re-added

File: modules/p2/ding_list/plugins/content_types/list_data.inc [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+  'category' => t('Dinglist'),
+  'render callback' => 'ding_list_list_data_content_type_render',
+  'required context' => new ctools_context_required(t('Ding list'), 'ding_list_list'),
+);
+
+/**
+ * Implements hook_content_type_render().
+ */
+function ding_list_list_data_content_type_render($subtype, $conf, $panel_args, $context) {
+  global $user;
```
[the remark this answers is not in the document]
Comment (b14sha) [reply modules/p2/ding_list/plugins/content_types/list_data.inc]: The whole list_data has been rewritten, and the $user var is omitted

File: themes/ddbasic/sass/p2/p2-view.scss (hunk 8 above)
Comment (b14sha) [reply themes/ddbasic/sass/p2/p2-view.scss]: Has been moved

File: modules/p2/ding_list/plugins/content_types/list_data.inc [hunk 12]
```diff
@@ -0,0 +1,118 @@
+<?php
+
+/**
+ * @file
+ * Information about list followers pane.
+ */
+
+$plugin = array(
+  'title' => t('Ding list data'),
```
[the remark this answers is not in the document]
Comment (kasperg) [reply modules/p2/ding_list/plugins/content_types/list_data.inc]: :+1: Thank for updating the doumentation.

File: themes/ddbasic/sass/components/ding-carousel.scss (hunk 6 above)
Comment (kasperg) [reply themes/ddbasic/sass/components/ding-carousel.scss]: Do you have a link to this? [I cannot see it](https://github.com/ding2/ding2/blob/master/modules/p2/ding_list/theme/theme.inc#L189-L201).

File: modules/p2/ding_list/plugins/content_types/list_data.inc (hunk 11 above)
[the remark this answers is not in the document]
Comment (kasperg) [reply modules/p2/ding_list/plugins/content_types/list_data.inc]: :+1:

File: modules/p2/ding_list/plugins/content_types/list_data.inc [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+        'ding-list-data--info',
+      ),
+    ),
+  );
+
+  $block->content['info']['user'] = array(
+    '#type' => 'container',
+    '#attributes' => array(
+      'class' => array(
+        'list-info',
```
[the remark this answers is not in the document]
Comment (kasperg) [reply modules/p2/ding_list/plugins/content_types/list_data.inc]: As far as I can tell this has not been addressed.

File: modules/p2/ding_list/plugins/content_types/list_data.inc [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+      '#label' => t('Number of followers'),
+      '#value' => format_plural($followers_count, '1 user following this list', '@count users following this list'),
+    );
+  }
+
+  $block->content['info']['follow'] = array(
+    '#type' => 'container',
+    '#attributes' => array(
+      'class' => array(
+        'list-buttons',
```
[the remark this answers is not in the document]
Comment (kasperg) [reply modules/p2/ding_list/plugins/content_types/list_data.inc]: As far as I can tell this has not been addressed.

File: modules/p2/ding_list/plugins/content_types/private_share.inc [hunk 15]
```diff
@@ -35,17 +35,19 @@ function ding_list_private_share_content_type_render($subtype, $conf, $panel_arg
...
       foreach ($conf['communities'] as $community) {
         $title = ucfirst($community);
-        $output->content .= l(
-          t('Share on @community', array('@community' => $title)),
-          'ding-list/private-share-message/' . $list->getId() . '/' . $community,
-          array(
-            'attributes' => array(
-              'class' => array(
-                'sharer-button', 'sharer-' . $community, 'use-ajax',
+        if ($title != '0') {
```
[the remark this answers is not in the document]
Comment (kasperg) [reply modules/p2/ding_list/plugins/content_types/private_share.inc]: As far as I can tell this has not been addressed.

File: modules/p2/ding_list/theme/theme.inc (hunk 1 above)
Comment (kasperg) [reply modules/p2/ding_list/theme/theme.inc]: As far as I can tell this has not been addressed.

File: modules/p2/ding_sharer/ding_sharer.module (hunk 4 above)
Comment (kasperg) [reply modules/p2/ding_sharer/ding_sharer.module]: Are you sure that you that this works right? I would expect that ` $variables['type']` refers to the value of the option ("normal") - not the textual version ("Normal"). The only place I would expect to see "Normal" would be in the textual version. Now it is spreading.

File: themes/ddbasic/sass/p2/p2-list.scss (hunk 7 above)
Comment (kasperg) [reply themes/ddbasic/sass/p2/p2-list.scss]: :+1:

File: themes/ddbasic/sass/p2/p2-view.scss (hunk 8 above)
Comment (kasperg) [reply themes/ddbasic/sass/p2/p2-view.scss]: :+1:

File: modules/p2/ding_list/plugins/content_types/list_data.inc [hunk 16]
```diff
@@ -0,0 +1,118 @@
...
+    $list_type = $list->getType();
+    if (isset($list_information[$list_type])) {
+      $list_information = $list_information[$list_type];
+      if (isset($list_information['_data']['description'])) {
+        $note = $list_information['_data']['description'];
+      }
+    }
+  }
+
+  $block->content['note'] = array(
```
[the remark this answers is not in the document]
Comment (kasperg) [reply modules/p2/ding_list/plugins/content_types/list_data.inc]: :+1:

File: modules/p2/ding_sharer/ding_sharer.module (hunk 5 above)
Comment (kasperg) [reply modules/p2/ding_sharer/ding_sharer.module]: :+1:

**Comment by kasperg** [comment]
I have gone through your followup. All the places where we agree I have resolved the conversation.

There are still 8 comments that are unresolved. Note that even though a comment is outdated it may still be relevant.

File: modules/p2/ding_list/plugins/content_types/list_data.inc (hunk 13 above)
[the remark this answers is not in the document]
Comment (b14sha) [reply modules/p2/ding_list/plugins/content_types/list_data.inc]: The markup has been moved to tje dong-list-data.tpl.php file, and the class-name changed to ding-list-info

File: modules/p2/ding_list/plugins/content_types/private_share.inc (hunk 15 above)
[the remark this answers is not in the document]
Comment (b14sha) [reply modules/p2/ding_list/plugins/content_types/private_share.inc]: Has been changed

File: modules/p2/ding_list/plugins/content_types/list_data.inc (hunk 14 above)
[the remark this answers is not in the document]
Comment (b14sha) [reply modules/p2/ding_list/plugins/content_types/list_data.inc]: The markup has been moved to ding-list-data.tpl.php file, and the class-name changed to ding-list-buttons

File: modules/p2/ding_list/theme/theme.inc (hunk 1 above)
Comment (b14sha) [reply modules/p2/ding_list/theme/theme.inc]: Have added the ding-list-list__owner class, and used the same naming as the existing classes. Have changed the naming of ding-list-data--info, to ding-list-data__info to use same naming convention.

File: modules/p2/ding_list/theme/theme.inc (hunk 2 above)
Comment (b14sha) [reply modules/p2/ding_list/theme/theme.inc]: Have added the ding-list-list__guest class, and used the same naming as the existing classes. Have changed the naming of ding-list-data--info, to ding-list-data__info to use same naming convention

File: modules/p2/ding_list/theme/theme.inc (hunk 3 above)
Comment (b14sha) [reply modules/p2/ding_list/theme/theme.inc]: Is changed

File: modules/p2/ding_sharer/ding_sharer.module (hunk 4 above)
Comment (b14sha) [reply modules/p2/ding_sharer/ding_sharer.module]: Youre are right! Thanx for clarifiyng!

File: themes/ddbasic/sass/components/ding-carousel.scss (hunk 6 above)
Comment (b14sha) [reply themes/ddbasic/sass/components/ding-carousel.scss]: Added here:
https://github.com/ding2/ding2/pull/1213/commits/9444088b70a133f4d8410979cdce6dbdebcaff80#diff-32b73877212e335408ce681993ee780dR196

**Comment by JrgenGNielsen** [comment]
review comments are resolved. 
However, there are merge conflicts, and needs a rebase. This may also resolve the failing behats test.

**Comment by JrgenGNielsen** [comment]
reviewed and approved
